### PR TITLE
Fix logo animation stuttering when entering mode from initial state

### LINKED ISSUE #4792: osu! "full size" logo has weird position when entering SongSelect
**Describe the bug:** The big version of the osu!logo has a weird transition when entering SongSelect without getting its "small" version.

**Screenshots or videos showing encountered issue:** https://streamable.com/0xbpi

**osu!lazer version:** 2019.514.0

**Logs:** N/A

### PR DESCRIPTION
**Pull request by nyquillerium**
Closes https://github.com/ppy/osu/issues/4792

The issue was caused by the immediate 0 duration tracking of the `OsuLogo` when starting tracking of `ButtonSystem`, and the fact that in the first place the initial location of the `Facade` is not centered on the logo itself, but where the `ButtonSystem` should be. 

To resolve this, I made the logo track over the same length of the `ButtonSystem` suspending transform. This modifies the tracking animation slightly, but I think it looks fine. Here's a preview using a modified `TestCaseButtonSystem`:

https://streamable.com/m380g

Edit: The alternative would be to fade out on the spot in this scenario without tracking, not sure if that would be better.

### CODE REVIEW

File: osu.Game/Screens/Menu/ButtonSystem.cs [hunk 1]
```diff
@@ -332,7 +332,8 @@ private void updateLogoState(ButtonSystemState lastState = ButtonSystemState.Ini
                     break;
 
                 case ButtonSystemState.EnteringMode:
-                    logoTrackingContainer.StartTracking(logo, 0, Easing.In);
+                    // When coming from the Initial (untracked) state, interpolate to the tracking position over a brief duration instead of tracking immediately.
```
Comment (peppy) [inline osu.Game/Screens/Menu/ButtonSystem.cs]: comment explains code 1:1 and therefore unnecessary.

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
Was there supposed to be a test added for this? How do I check it's working?

**Comment by nyquillerium** [comment]
I modified `TestSceneButtonSystem` like this to visualize, but it seemed kinda messy to have in the test case
```
foreach (var s in Enum.GetValues(typeof(ButtonSystemState)).OfType<ButtonSystemState>().Skip(1))
  AddStep($"State to {s}", () =>
  {
      buttons.State = s;

      if (buttons.State == ButtonSystemState.EnteringMode)
      {
        buttons.FadeOut(400, Easing.InSine);
        buttons.MoveTo(new Vector2(-800, 0), 400, Easing.InSine);
        logo.FadeOut(300, Easing.InSine)
         .ScaleTo(0.2f, 300, Easing.InSine);
      }
      else
      {
          buttons.FadeIn(400, Easing.OutQuint);
          buttons.MoveTo(new Vector2(0), 400, Easing.OutQuint);
          logo.FadeColour(Color4.White, 100, Easing.OutQuint);
          logo.FadeIn(100, Easing.OutQuint);
      }
  });
```

**Comment by peppy** [comment]
I think it's fine to have as its own test method, as we do elsewhere. If it's not there as a test the potential for regression is higher (even if not automated).

**Comment by peppy** [comment]
I split out the test into its own test method (as I proposed) which may have revealed another flaw in logic currently: the animation is different if it is executed earlier.

Not a regression, but worth noting.

**Comment by nyquillerium** [comment]
Ah, I didn't understand the initial proposal.

The 400 is a length specified locally in `OnSuspending` of `MainMenu`, and the transform is applied screen-wide. https://github.com/ppy/osu/blob/0ec01f7b0fa5ca4ae8b4f3a76182ff5be2fdb493/osu.Game/Screens/Menu/MainMenu.cs#L174

**Comment by peppy** [comment]
please expose and use the constant so it's obvious, then.

File: osu.Game/Screens/Menu/ButtonSystem.cs [hunk 2]
```diff
@@ -332,7 +332,7 @@ private void updateLogoState(ButtonSystemState lastState = ButtonSystemState.Ini
                     break;
 
                 case ButtonSystemState.EnteringMode:
-                    logoTrackingContainer.StartTracking(logo, 0, Easing.In);
+                    logoTrackingContainer.StartTracking(logo, lastState == ButtonSystemState.Initial ? 400 : 0, Easing.InSine);
```
Comment (peppy) [inline osu.Game/Screens/Menu/ButtonSystem.cs]: .

**Review by peppy — APPROVED** [review]

**Pull request merged**